test/conformance/program/urProgramGetInfo.cpp

```diff
@@ -25,13 +25,31 @@ UUR_TEST_SUITE_P(
 
 TEST_P(urProgramGetInfoTest, Success) {
     auto property_name = getParam();
-    size_t property_size = 0;
     std::vector<char> property_value;
-    ASSERT_SUCCESS(
-        urProgramGetInfo(program, property_name, 0, nullptr, &property_size));
-    property_value.resize(property_size);
-    ASSERT_SUCCESS(urProgramGetInfo(program, property_name, property_size,
-                                    property_value.data(), nullptr));
+    if (property_name == UR_PROGRAM_INFO_BINARIES) {
+        size_t binary_sizes_len = 0;
+        ASSERT_SUCCESS(urProgramGetInfo(program, UR_PROGRAM_INFO_BINARY_SIZES,
+                                        0, nullptr, &binary_sizes_len));
+        // Due to how the fixtures + env are set up we should only have one
+        // device associated with program, so one binary.
+        ASSERT_EQ(binary_sizes_len / sizeof(size_t), 1);
+        size_t binary_sizes[1] = {binary_sizes_len};
+        ASSERT_SUCCESS(urProgramGetInfo(program, UR_PROGRAM_INFO_BINARY_SIZES,
+                                        binary_sizes_len, binary_sizes,
+                                        nullptr));
+        std::vector<char> binary(binary_sizes[0]);
+        char *binaries[1] = {binary.data()};
```

Comment:
I think we could check that the pointer returned is the same as the binary pointer that we initialized the program with

Reply:
This is for querying out the device binary, I'll add something like that for `GET_IL` in https://github.com/oneapi-src/unified-runtime/pull/1196. I started adding a bit to this test where we feed the binary that was returned back into `urProgramCreateWithBinary` and then query the binary out of the new program and verify it against the binary we passed in, but then it occurred to me that `GET_PROGRAM_BINARIES` is very implementation dependant. The opencl equivalent of the query says the binary you get out is completely implementation defined, so as far as we know a given CL implementation might create a program with a binary that contains slightly different bits if it was created with `CreateWithBinary` and then built with `ProgramBuild`. There's no guarantee of consistency.

Reply:
That make sense. Thanks for clarification.